File: cpp/pascals_triangle_ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> result;
        for (int row=0; row!=rowIndex+1; ++row)
        {
            result.push_back(1);
            for (vector<int>::reverse_iterator iter=result.rbegin(); iter!=result.rend(); ++iter)
            {
                if (iter!=result.rbegin() && iter!=result.rend()-1)
                    *iter = *iter + *(iter+1);
            }
        }
        return result;
    }
};

File: python/pascals_triangle_ii.py
class Solution:
    # @return a list of integers
    def getRow(self, rowIndex):
        result=[1]
        for row in range(rowIndex+1):
            for index in range(row):
                if index == 0:
                    result.append(1)
                else:
                    result[row-index] += result[row-index-1]
        return result